### PR TITLE
adds warning for mismatched class member accessibility

### PR DESCRIPTION
**Pull request by georgejecook**

### CODE REVIEW

File: src/validators/ClassValidator.ts [hunk 1]
```diff
@@ -223,6 +224,24 @@ export class BsClassValidator {
                                 range: member.range
                             });
                         }
+
+                        //child member has different visiblity
+                        if (
+                            //is a method
+                            isClassMethodStatement(member) &&
```
Comment (TwitchBronBron) [inline src/validators/ClassValidator.ts:231]: Why only class methods? Fields should also be included in this. TypeScript applies this restriction to both methods and fields.

**Review by TwitchBronBron — CHANGES_REQUESTED** [review]
Looks good. I asked for a few small changes.

Also, the github actions build is failing right now because coveralls (our coverage tool) has a server-side error. I'll keep an eye on that.

File: src/validators/ClassValidator.ts (hunk 1 above)
Comment (georgejecook) [reply src/validators/ClassValidator.ts:231]: I did this as we don't support overriding field yet, do we?

**Comment by TwitchBronBron** [comment]
Looks good. Thanks!

**Pull request merged**